sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py

```diff
@@ -323,6 +386,108 @@ def __repr__(self):
+    """
+
+    def __init__(self, **kwargs):
+        self.text = kwargs.get("text", None)
+        self.category = kwargs.get("category", None)
+        self.subcategory = kwargs.get("subcategory", None)
+        self.offset = kwargs.get("offset", None)
+        self.length = kwargs.get("length", None)
+        self.confidence_score = kwargs.get("confidence_score", None)
+        self.links = kwargs.get("links", [])
+
+    @classmethod
+    def _from_generated(cls, healthcare_entity):
+        return cls(
+            text=healthcare_entity.text,
+            category=healthcare_entity.category,
+            subcategory=healthcare_entity.subcategory,
+            offset=healthcare_entity.offset,
+            length=healthcare_entity.length,
+            confidence_score=healthcare_entity.confidence_score,
+            links=[HealthcareEntityLink(id=l.id, data_source=l.data_source) for l in healthcare_entity.links] if healthcare_entity.links else None
+        )
+
+    def __repr__(self):
+        return "HealthcareEntity(text={}, category={}, subcategory={}, offset={}, length={}, confidence_score={},\
+        links={})".format(
+            self.text,
+            self.category,
+            self.subcategory,
+            self.offset,
+            self.length,
+            self.confidence_score,
+            repr(self.links)
+        )
+
+
+class HealthcareRelation(DictMixin):
+    """
+    HealthcareRelation contains information describing a relationship between two entities found in text.
+
+    :ivar str type: The type of relation, such as DosageOfMedication or FrequencyOfMedication, etc.
+    :ivar bool is_bidirectional: Boolean value indicating that the relationship between the two entities is
+        bidirectional.  If true the relation between the entities is bidirectional, otherwise directionality
+        is source to target.
+    ivar source: A reference to an extracted Healthcare entity representing the source of the relation.
+    vartype source: ~azure.ai.textanalytics.HealthcareEntity
+    ivar target: A reference to an extracted Healthcare entity representing the target of the relation.
+    vartype target: ~azure.ai.textanalytics.HealthcareEntity
+    """
+
+    def __init__(self, **kwargs):
+        self.relation_type = kwargs.get("relation_type", None)
+        self.is_bidirectional = kwargs.get("is_bidirectional", None)
+        self.source = kwargs.get("source", None)
+        self.target = kwargs.get("target", None)
+
+    @classmethod
+    def _from_generated(cls, healthcare_relation, source_entity, target_entity):
+        return cls(
+            relation_type=healthcare_relation.relation_type,
+            is_bidirectional=healthcare_relation.bidirectional,
+            source=source_entity,
+            target=target_entity
+        )
+
+    def __repr__(self):
+        return "HealthcareRelation(relation_type={}, is_bidirectional={}, source={}, target={})".format(
```

Comment:
nit: i think it's a good idea to just include [:1024] for every repr, so they consistently don't go over 1024 characters

Reply:
I added this to __repr__ for HealthcareRelation, HealthcareEntity, and HealthcareEntityLink where it was missing.